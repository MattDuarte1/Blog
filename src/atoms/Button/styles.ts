import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme, disabled }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 20px 23px 19px;
    gap: 10px;
    border: none;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};

    min-width: 145px;
    height: 67px;
    font-weight: 700;
    color: #ffff;

    background: ${theme.colors.blue};
    border-radius: 8px;
    transition: filter 0.4s ease;

    filter: ${disabled && 'brightness(0.7)'};

    &:hover {
      filter: ${!disabled && 'brightness(1.1)'};
    }
  `}
`;
