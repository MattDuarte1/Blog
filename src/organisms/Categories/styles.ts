import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  ContainerSize: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        padding: ${theme.spacings.medium};
      }
    `,
  },
  CategoryBodySize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        div {
          width: 100%;
        }
      }
    `,
  },
};

export const Container = styled.section`
  ${({ theme }) => css`
    display: block;
    padding: ${theme.spacings.xxlarge};
    background-color: ${theme.colors.white};
    width: 100%;
    height: 100%;

    ${BreakPoints.ContainerSize['medium'](theme)}
  `}
`;

export const CategoryHeader = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 2rem;
`;

export const CategoryBody = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    min-width: 100%;
    height: 100%;
    gap: 37px;

    ${BreakPoints.CategoryBodySize['small'](theme)}
  `}
`;
