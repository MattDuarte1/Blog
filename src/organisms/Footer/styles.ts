import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    width: 100%;
    padding: 5rem;

    box-shadow: 1px 0 1rem 0 rgba(0, 0, 0, 0.2);

    display: flex;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      gap: 4rem;
      padding: 5rem;
    }
  `}
`;

export const Content = styled.aside`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.small};

    flex: 1;
  `}
`;

export const SocialMediaContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: 2rem;
    justify-content: center;
    width: 40%;
    a {
      color: ${theme.colors.black};
    }

    text-decoration: none;
    color: ${theme.colors.black};
    svg {
      cursor: pointer;
      font-size: 25px;
      transition: all 0.4s ease;

      &:hover {
        transform: scale(1.1);
      }
    }
  `}
`;
