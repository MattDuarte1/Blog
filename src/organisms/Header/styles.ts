import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 86px;
    background-color: ${theme.colors.white};
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.05));

    position: relative;
  `}
`;
