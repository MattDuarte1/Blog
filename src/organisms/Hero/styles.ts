import styled, { css, DefaultTheme, keyframes } from 'styled-components';

const BreakPoints = {
  Container: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteLarge} {
        padding: ${theme.spacings.huge} ${theme.spacings.medium};
      }
    `,
  },
  DescriptionSize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        margin-top: 20px;
        font-size: ${theme.font.sizes.xsmall};
        line-height: normal;
      }
    `,
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        margin-top: 40px;
        font-size: ${theme.font.sizes.small};
        line-height: normal;
      }
    `,
  },
  AsideRight: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        img {
          width: 280px;
          height: 280px;
        }
      }
    `,
  },
};

const cursorEffect = (theme: DefaultTheme) => keyframes`
  0% {
    background-color: ${theme.colors.black};
  }
  100% {
    background-color: ${theme.colors.white};
  }
`;

export const Container = styled.section`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xhuge} ${theme.spacings.medium};
    min-height: 700px;
    display: flex;
    background-color: ${theme.colors.white};
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;

    ${BreakPoints.Container['medium'](theme)}
  `}
`;

export const AsideLeft = styled.aside`
  ${({ theme }) => css`
    max-height: 400px;
    max-width: 664px;
    padding: 25px
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 30px;

    span {
      color: transparent;
      width: 5px;
      height: 100%;
      width: 2px;
      animation: ${cursorEffect(theme)} 1s infinite;
    }
  `}
`;

export const AsideRight = styled(AsideLeft)`
  ${({ theme }) => css`
    justify-content: center;
    display: block;

    ${BreakPoints.AsideRight['medium'](theme)}
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    padding-left: 15px;
    margin-top: 25px;
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 400;
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.darkGrey};
    line-height: 25px;
    margin-left: 20px;
    position: relative;

    &::before {
      content: '';
      position: absolute;
      left: 0px;
      background-color: ${theme.colors.black};
      width: 5px;
      height: 100%;
    }

    ${BreakPoints.DescriptionSize['small'](theme)}
    ${BreakPoints.DescriptionSize['medium'](theme)}
  `}
`;
