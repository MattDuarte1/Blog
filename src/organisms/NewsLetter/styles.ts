import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  Content: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      > h1 {
        font-size: ${theme.font.sizes.medium};
      }

      @media ${theme.media.lteSmall} {
        > h1 {
          font-size: ${theme.font.sizes.small};
        }
        > span {
          font-size: ${theme.font.sizes.xsmall};
        }
      }
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 58px;
    gap: 10px;
    background-color: ${theme.name === 'inverted'
      ? theme.colors.white
      : theme.colors.lightGrey2};
    width: 100%;
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 0px;
    gap: 30px;

    max-width: 696px;

    ${BreakPoints.Content(theme)}
  `}
`;

export const Form = styled.form`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 20px;
  width: 100%;
`;

export const InputForm = styled.input`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
  padding: 21px 21px 21px 25px;

  width: 300px;
  height: 67px;

  /* White */

  background: #fffffe;
  border-radius: 8px;
  border: none;

  &:focus {
    box-shadow: 0 0 0 0;
    outline: 0;
  }
`;
