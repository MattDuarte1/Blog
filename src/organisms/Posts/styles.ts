import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreakPoints = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      padding: 3.5rem;
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 4.4rem;
    background-color: ${theme.colors.white};
    ${MediaBreakPoints.small(theme)}
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: 1.4rem;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(26rem, 2fr));
    gap: ${theme.spacings.large};

    @media (max-width: 1065px) {
      grid-template-columns: repeat(auto-fill, minmax(27rem, 1fr));
    }

    @media (max-width: 705px) {
      justify-content: center;
      grid-template-columns: 0fr;
    }
  `}
`;
