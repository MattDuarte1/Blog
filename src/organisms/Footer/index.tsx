import * as Styled from './styles';
import { forwardRef } from 'react';
import Logo from '@/atoms/Logo';
import Heading from '@/atoms/Heading';
import AnchorLink from '@/atoms/Anchor';
import { RiGithubFill, RiLinkedinLine } from 'react-icons/ri';

const Footer = forwardRef<HTMLDivElement>(() => {
  return (
    <Styled.Container id="footer">
      <Styled.Content>
        <Logo />
        <Heading color="black" as="h6" size="xsmall">
          Criado por Mateus Duarte
        </Heading>
        <Styled.SocialMediaContent>
          <AnchorLink
            href="https://github.com/MattDuarte1"
            newTab
            rel="noreferrer"
          >
            <RiGithubFill />
          </AnchorLink>
          <AnchorLink
            href="https://www.linkedin.com/in/mateus-duarte-aa2a1b1b3/"
            newTab
            rel="noreferrer"
          >
            <RiLinkedinLine />
          </AnchorLink>
        </Styled.SocialMediaContent>
        <AnchorLink
          href={'/api/preview?pass=9214984e8f485ds4f8ds4f8438r4w98f4s6d541c'}
        >
          <Heading color="black" size="xsmall">
            Ativar/Desativar Modo Real-Time
          </Heading>
        </AnchorLink>
      </Styled.Content>
    </Styled.Container>
  );
});

Footer.displayName = 'Footer component';

export default Footer;
