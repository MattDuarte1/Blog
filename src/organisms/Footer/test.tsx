import Footer from '.';
import { renderTheme } from '@/styles/render-theme';
import { screen } from '@testing-library/react';

describe('<Footer>', () => {
  it('Should render a Footer Component', () => {
    renderTheme(<Footer />);
    const footer = screen.getByRole('contentinfo');

    expect(footer).toBeInTheDocument();
  });

  it('Should match snapshot footer component', () => {
    renderTheme(<Footer />);
    const footer = screen.getByRole('contentinfo');

    expect(footer).toMatchSnapshot();
  });
});
