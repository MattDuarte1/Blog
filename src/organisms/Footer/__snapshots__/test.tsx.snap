// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> Should match snapshot footer component 1`] = `
.c0 {
  background-color: #FFFFFE;
  width: 100%;
  padding: 5rem;
  box-shadow: 1px 0 1rem 0 rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2.156rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40%;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #2B2C34;
}

.c4 a {
  color: #2B2C34;
}

.c4 svg {
  cursor: pointer;
  font-size: 25px;
  -webkit-transition: all 0.4s ease;
  transition: all 0.4s ease;
}

.c4 svg:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c2 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 3.576rem;
  line-height: 45px;
  position: relative;
}

.c2::first-letter {
  color: #6246EA;
}

.c2::after {
  content: '.blog';
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 2.156rem;
  color: #6246EA;
}

.c3 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

.c5 {
  cursor: pointer;
  color: #2B2C34;
}

@media (max-width:700px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 4rem;
    padding: 5rem;
  }
}

<footer
  class="c0"
  id="footer"
>
  <aside
    class="c1"
  >
    <div
      class="c2"
    >
      BlogZin
    </div>
    <h6
      class="c3"
    >
      Criado por Mateus Duarte
    </h6>
    <div
      class="c4"
    >
      <a
        class="c5"
        href="https://github.com/MattDuarte1"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.475 2 2 6.475 2 12a9.994 9.994 0 0 0 6.838 9.488c.5.087.687-.213.687-.476 0-.237-.013-1.024-.013-1.862-2.512.463-3.162-.612-3.362-1.175-.113-.288-.6-1.175-1.025-1.413-.35-.187-.85-.65-.013-.662.788-.013 1.35.725 1.538 1.025.9 1.512 2.338 1.087 2.912.825.088-.65.35-1.087.638-1.337-2.225-.25-4.55-1.113-4.55-4.938 0-1.088.387-1.987 1.025-2.688-.1-.25-.45-1.275.1-2.65 0 0 .837-.262 2.75 1.026a9.28 9.28 0 0 1 2.5-.338c.85 0 1.7.112 2.5.337 1.912-1.3 2.75-1.024 2.75-1.024.55 1.375.2 2.4.1 2.65.637.7 1.025 1.587 1.025 2.687 0 3.838-2.337 4.688-4.562 4.938.362.312.675.912.675 1.85 0 1.337-.013 2.412-.013 2.75 0 .262.188.574.688.474A10.016 10.016 0 0 0 22 12c0-5.525-4.475-10-10-10z"
            />
          </g>
        </svg>
      </a>
      <a
        class="c5"
        href="https://www.linkedin.com/in/mateus-duarte-aa2a1b1b3/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 9.55C12.917 8.613 14.111 8 15.5 8a5.5 5.5 0 0 1 5.5 5.5V21h-2v-7.5a3.5 3.5 0 0 0-7 0V21h-2V8.5h2v1.05zM5 6.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm-1 2h2V21H4V8.5z"
            />
          </g>
        </svg>
      </a>
    </div>
    <a
      class="c5"
      href="/api/preview?pass=9214984e8f485ds4f8ds4f8438r4w98f4s6d541c"
      target="_self"
    >
      <h1
        class="c3"
      >
        Ativar/Desativar Modo Real-Time
      </h1>
    </a>
  </aside>
</footer>
`;
